src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -1355,5 +1374,120 @@ private void VerifyNotepadMainWindowTitle(Process process, string filename)
+                // to its output stream so we can read them.
+                p = CreateProcess(arg =>
+                {
+                    Console.Write(File.ReadAllText(arg));
+                    return RemoteExecutor.SuccessExitCode;
+                }, testFileUncPath);
+                p.StartInfo.StandardOutputEncoding = Encoding.UTF8;
+                p.StartInfo.RedirectStandardOutput = true;
+                p.StartInfo.LoadUserProfile = false;
+                p.StartInfo.UseCredentialsForNetworkingOnly = true;
+                p.StartInfo.UserName = username;
+                p.StartInfo.PasswordInClearText = password;
+
+                workingDirectory = string.IsNullOrEmpty(p.StartInfo.WorkingDirectory)
+                    ? Directory.GetCurrentDirectory()
+                    : p.StartInfo.WorkingDirectory;
+
+                if (PlatformDetection.IsNotWindowsServerCore) // for this particular Windows version it fails with Attempted to perform an unauthorized operation (#46619)
+                {
+                    // ensure the new user can access the .exe and the temp file (otherwise you get Access is denied exception)
+                    SetAccessControl(accountWithoutFilePermission.AccountName, p.StartInfo.FileName, workingDirectory, add: true);
+                    SetAccessControl(username, testFilePath, Path.GetDirectoryName(testFilePath), add: true);
+                }
+
+                try
+                {
+                    hasStarted = WindowsIdentity.RunImpersonated(
+                        accountWithoutFilePermission.AccountTokenHandle,
+                        p.Start);
+                    string output = p.StandardOutput.ReadToEnd();
+                    Assert.True(p.WaitForExit(WaitInMS));
+
+                    Assert.Equal(testFileContent, output);
+                }
+                catch (Win32Exception ex) when (ex.NativeErrorCode == ERROR_SHARING_VIOLATION)
+                {
+                    throw new SkipTestException($"{p.StartInfo.FileName} has been locked by some other process");
+                }
+
+                Assert.Equal(accountWithoutFilePermission.AccountName.Split('\\').LastOrDefault(), Helpers.GetProcessUserName(p));
+            }
+            finally
+            {
+                if (hasStarted)
+                {
+                    p.Kill();
+
+                    Assert.True(p.WaitForExit(WaitInMS));
+                }
+
+                if (PlatformDetection.IsNotWindowsServerCore)
+                {
+                    // remove the access
+                    SetAccessControl(accountWithoutFilePermission.AccountName, p.StartInfo.FileName, workingDirectory, add: false);
+                    SetAccessControl(username, testFilePath, Path.GetDirectoryName(testFilePath), add: false);
+                }
+
+                Assert.Equal(Interop.ExitCodes.NERR_Success, Interop.NetUserDel(null, username));
```

Comment:
it would be great if you could refactor this test and `TestUserCredentialsPropertiesOnWindows` and re-use common logic. It could be something like this:

```cs
private static void CreateUserAndExecute(Process process, string userName, string password)
```